### DISCUSSION TITLE
Dynamic routes with rewrites on vercel

### DISCUSSION THREAD

**Discussion by jakeorr** [vercel/next.js#29237] [Help]
Hello! I'm using URL rewrites to adjust URL structure in my Next.js 11 project. It works great in development, but when it is deployed on vercel and I try to access a dynamic route through the rewrites I get a 404. Here is my rewrite config:

```js
module.exports = {
  reactStrictMode: true,
  async rewrites() {
    return [
      {
        source: '/:path*',
        has: [
          {
            type: 'query',
            key: 'fid',
            value: `(?<fid>.*)`,
          }
        ],
        destination: `/:fid/:path*`,
      },
    ]
  }
}
```

I have a dynamic route at `pages/a-funnel/orders/[orderId].js`. When I try to access `/orders/1234?fid=a-funnel` I get a 404, but only on the version deployed on vercel.

I have seen a number of similar issues ([example](https://github.com/vercel/next.js/discussions/13347)), but haven't yet found a solution for this.

Here is a minimal reproduction: https://github.com/uncoil/dynamic-route-rewrite/tree/minimal
You can see a vercel deployment of it at: https://dynamic-route-rewrite-4dwj7txal-uncoil.vercel.app/

**Comment by codeXLinkX** [comment]
is there an answer for this?
